includes/Locale.php

```diff
@@ -0,0 +1,214 @@
+		'ja_JP',
+		'ja_KS',
+		'jv_ID',
+		'ka_GE',
+		'kk_KZ',
+		'km_KH',
+		'kn_IN',
+		'ko_KR',
+		'ku_TR',
+		'lt_LT',
+		'lv_LV',
+		'mg_MG',
+		'mk_MK',
+		'ml_IN',
+		'mn_MN',
+		'mr_IN',
+		'ms_MY',
+		'mt_MT',
+		'my_MM',
+		'nb_NO',
+		'ne_NP',
+		'nl_BE',
+		'nl_NL',
+		'nn_NO',
+		'or_IN',
+		'pa_IN',
+		'pl_PL',
+		'ps_AF',
+		'pt_BR',
+		'pt_PT',
+		'qz_MM',
+		'ro_RO',
+		'ru_RU',
+		'rw_RW',
+		'sc_IT',
+		'si_LK',
+		'sk_SK',
+		'sl_SI',
+		'so_SO',
+		'sq_AL',
+		'sr_RS',
+		'sv_SE',
+		'sw_KE',
+		'sz_PL',
+		'ta_IN',
+		'te_IN',
+		'tg_TJ',
+		'th_TH',
+		'tl_PH',
+		'tr_TR',
+		'tz_MA',
+		'uk_UA',
+		'ur_PK',
+		'uz_UZ',
+		'vi_VN',
+		'zh_CN',
+		'zh_HK',
+		'zh_TW',
+	];
+
+
+	/**
+	 * Gets a list of locales supported by Facebook.
+	 *
+	 * @link https://developers.facebook.com/docs/messenger-platform/messenger-profile/supported-locales/
+	 * If the Locale extension is not available, will attempt to match locales to WordPress available language names.
+	 *
+	 * @since 2.2.0-dev.1
+	 *
+	 * @param bool $with_labels whether to output a list of locale identifiers or an associative array of identifiers and labels
+	 * @return array
+	 */
+	public static function get_supported_locales( $with_labels = true ) {
+
+		$locales = [];
+
+		if ( class_exists( 'Locale' ) ) {
+
+			foreach ( self::$supported_locales as $locale ) {
+
+				if ( $name = \Locale::getDisplayName( $locale, substr( $locale, 0, 2 ) ) ) {
+
+					$locales[ $locale ] = ucfirst( $name );
+				}
+			}
+
+		} else {
+
+			include_once( ABSPATH . '/wp-admin/includes/translation-install.php' );
+
+			$translations = wp_get_available_translations();
+
+			foreach ( self::$supported_locales as $locale ) {
+
+				if ( isset( $translations[ $locale ]['native_name'] ) ) {
+
+					$locales[ $locale ] = $translations[ $locale ]['native_name'];
+
+				} else { // generic match e.g. <it>_IT, <it>_CH (any language in the the <it> group )
+
+					$matched_locale = substr( $locale, 0, 2 );
+
+					if ( isset( $translations[ $matched_locale ]['native_name'] ) ) {
+						$locales[ $locale ] = $translations[ $matched_locale ]['native_name'];
+					}
+				}
+			}
+
+			// always include US English
+			$locales['en_US'] = _x( 'English (United States)', 'language', 'facebook-for-woocommerce' );
+		}
+
+		/**
+		 * Filters the locales supported by Facebook Messenger.
+		 *
+		 * @since 1.10.0
+		 *
+		 * @param array $locales locales supported by Facebook, in $locale => $name format
+		 */
+		$locales = (array) apply_filters( 'wc_facebook_messenger_supported_locales', array_unique( $locales ) );
+
+		natcasesort( $locales );
+
+		return $with_labels ? $locales : array_keys( $locales );
+	}
+
+
+	/**
+	 * Determines if a locale is supported by Facebook.
+	 *
+	 * @since 2.2.0-dev.1
+	 *
+	 * @param string $locale a locale identifier
+	 * @return bool
+	 */
+	public static function is_supported_locale( $locale ) {
+
+		return in_array( $locale, self::get_supported_locales( false ), false );
```

Comment:
@unfulvio I think adding a `$with_labels` boolean parameter decreases readability for the code that uses `get_supported_locales()` and increases the responsibility of the method. Considering that we are only using the boolean attribute in `is_supported_locale()`, what do you think about removing that parameter and moving the call to `array_keys()` here:

```suggestion
		return in_array( $locale, array_keys( self::get_supported_locales() ), false );
```

Also, what's the reason to use `$strict = false` explicitly here?

Reply:
none really, used to be array_key_exists and I suppose we can restore to that given your suggestion to ignore a bool parameter in the called method

Reply:
https://github.com/facebookincubator/facebook-for-woocommerce/pull/1667/commits/538ca2c0c8d92ab7b1e68068590222e3b0955ca9